src/presentational/Armeggeddon/Armeggeddon.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect, useContext } from 'react';
+import FirebaseContext from '../../container/Firebase/Context'
+
+import Masthead from '../Components/Masthead/Masthead';
+import { Table, TableBody,TableHead, TableRow, TableCell } from '@material-ui/core';
+
+function RealtimeExample(){
+    const [state, setState] = useState([]);
+    const firebase = useContext(FirebaseContext);
+
+    useEffect(() => {
+        const unsubscribe = firebase.db.collection('president')
+            .onSnapshot( snapshot => {
+                const results = []
+                snapshot.forEach( doc => { 
+                    results.push(doc.data())
+                    
+                })
+                console.log(results)
+                setState(results)
+        })
+        return () => unsubscribe()
+    },[firebase]);
+
+    return (
+        <div>
+            <Masthead title={"Realtime"}/>
+            <Table>
+            <TableHead>
+                <TableRow>
+                    <TableCell>候選人</TableCell>
+                    <TableCell>得票數</TableCell>
+                </TableRow>
+            </TableHead>
+            <TableBody>
+                {state.map(candidate => (
+                <TableRow>
+                    <TableCell>{candidate.name}</TableCell>
+                    <TableCell>{candidate.vote}</TableCell>
+                </TableRow>
+                ))}
+            </TableBody>
+            </Table>
+        </div>
+    );
+}
+
+function RestfulExample() {
+    const [state, setState] = useState([]);
+    const firebase = useContext(FirebaseContext);
+
+    useEffect(() => {
+        firebase.db.collection('mayor_raw').get()
+            .then((snapshot) => {
+                const results = []
```

Comment:
This can be simpler
`const result = snapshot.map(d => ({id: d.id, ...d.data()});`

Reply:
It's odd because normally I would say you're right. But I will encounter this error: `TypeError: snapshot.map is not a function`

Somehow I read there is no map function provided in firebase SDK somewhere. I will try to find the document.

Reply:
It turns out snapshot is not an array. According to the [docs](https://firebase.google.com/docs/reference/js/firebase.firestore.QuerySnapshot):
`snapshot.docs` is the result array.

So this should change to
`const result = snapshot.docs.map(d => ({id: d.id, ...d.data()}));`